ecl/hql/hqlgram.y

```diff
@@ -8779,7 +8815,9 @@ simpleDataSet
                                 HqlExprArray args;
                                 unwindChildren(args, transform);
                                 transform.setown(createValue(no_newtransform, transform->getType(), args));
-                                $$.setExpr(createDataset(no_newkeyindex, dataset, createComma(record.getClear(), transform.getClear(), extra.getClear(), parser->getGpgSignature())));
+                                IHqlExpression *index = createDataset(no_newkeyindex, dataset, createComma(record.getClear(), transform.getClear(), extra.getClear(), parser->getGpgSignature()));
```

Comment:
minor scope for memory leak: If checkValidLookupFlag reports an error that then aborts.